feature/39_COMMENT_FUNCTIONALITY

Symphonie234 opened the pull request:
### Changes Made

- Added a functionality to add comments
- Added a functionality to like comments
- Added a new **Controller** file named **`CommentController`**
- Added **`Comment`** and **`CommentLike`** Models
- Added new routes in the **`web.php`** route file
- Added **migration** files for comments and likes on it

Symphonie234 committed:
functionality for commenting in the listing

Symphonie234 committed:
Merge branch 'develop' into feature/39_COMMENT_FUNCTIONALITY

Symphonie234 committed:
added a functionality to like and unlike a comment

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+    public function store(Request $request)
+    {
+        // Validate the request data
+        $validatedData = $request->validate([
```
make validation file

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+        ]);
+
+        // Create a new comment in the database
+        $comment = new Comment([
```
Please use register trait

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+        broadcast(new CommentPosted($comment))->toOthers();
+
+        // Redirect back to the specific listing page
+        return redirect()->route('listings.show', ['listing' => $request->input('listing_id')])
```
use message lang

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+    public function index($listingId)
+    {
+        // Replace 'Comment' with your actual Comment model
+        $comments = Comment::where('listing_id', $listingId)->latest()->get();
```
don't have return?

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+     * @param  int  $commentId
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function like(Request $request, $commentId)
```
$request parameter looks like not use

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+                $commentLike->delete();
+            }
+
+            return redirect()->back()->with('success', 'Comment unliked successfully');
```
use message lang

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,132 @@
+            }
+        }
+
+        // Redirect back to the page (the comment's parent listing) after unliking
```
use message lang

ChronoDevs commented on app/Http/Controllers/ListingController.php:
```diff
@@ -199,4 +203,32 @@ public function unlike(Listing $listing)
+        $user = auth()->user();
+
+        // Validate the request data for the comment
+        $validatedData = $request->validate([
```
make validation file

ChronoDevs commented on app/Http/Controllers/ListingController.php:
```diff
@@ -199,4 +203,32 @@ public function unlike(Listing $listing)
+        ]);
+
+        // Create a new comment
+        $comment = new Comment([
```
use register trait

ChronoDevs commented:
Please fix the conflict. Thank you

Symphonie234 committed:
Merge branch 'develop' into feature/39_COMMENT_FUNCTIONALITY

Symphonie234 committed:
Attended to the requested changes posted in the PR

Symphonie234 commented:
### Changes that have been addressed in the PR No. 1:

- Created two new **Requests** files named **`CommentStoreRequest.php`** and **`ListingStoreCommentRequest.php`**.
- Have the **`store`** method in the **CommentController** and **`storeComment`** method in the **`ListingController`** use the **`Register.php`** **Trait**.
- Created new messages in the **`lang/messages.php`** to be used in the **`CommentController`** methods.
- Fixed the conflict in the **`User`** **Model**.

Symphonie234 committed:
added a feature to show number of comments of the listing

Symphonie234 commented:
### New Change/s:

- Added a feature to show the number of comments the listing has in the **`listings/show.blade.php`**.

ChronoDevs commented on app/Events/CommentPosted.php:
```diff
@@ -0,0 +1,40 @@
+
+    public $comment;
+
+    public function __construct($comment)
```
Please add comment method above every function. Thank you

ChronoDevs commented on app/Http/Controllers/CommentController.php:
```diff
@@ -0,0 +1,141 @@
+
+class CommentController extends Controller
+{
+    public function __construct()
```
Please add comment method above every function. Thank you

Symphonie234 committed:
Attended to the requested changes posted in the PR No. 2

Symphonie234 commented:
### Changes that have been addressed in the PR No. 2:

- Added comments above the functions of the **Event** file **`CommentPosted.php`**.
- Added comments above the functions of the **Controller** file **`CommentController.php`**.
